hypothesis-python/tests/cover/test_lookup.py

```diff
@@ -540,6 +540,18 @@ def test_resolving_recursive_type():
     assert isinstance(st.builds(Tree).example(), Tree)
 
 
+class LinkedList:
+    def __init__(self, nxt: typing.Optional["LinkedList"] = None):
+        self.nxt = nxt
+
+    def __repr__(self):
+        return f"LinkedList({self.nxt})"
+
+
+def test_resolving_recursive_type_with_defaults():
+    assert isinstance(st.from_type(LinkedList).example(), LinkedList)
```

Comment:
It's used here, just a simplified version of `Tree` above, plus default arg. Maybe it should have a less functionality-sounding name